geopandas/io/tests/test_pickle.py

```diff
@@ -58,3 +59,18 @@ def test_round_trip_current(tmpdir, current_pickle_data):
         value.to_pickle(path)
         result = pd.read_pickle(path)
         assert_geodataframe_equal(result, value)
+        assert isinstance(result.has_sindex, bool)
+
+
+@pytest.mark.skipif(not compat.USE_PYGEOS, reason="requires pygeos to test #1745")
+def test_pygeos_switch(tmpdir):
+    geopandas.options.use_pygeos = False
+    gdf_crs = geopandas.GeoDataFrame(
+        {"a": [0.1, 0.2, 0.3], "geometry": [Point(1, 1), Point(2, 2), Point(3, 3)]},
+        crs="EPSG:4326",
+    )
+    path = str(tmpdir / "{}.pickle".format(gdf_crs))
+    gdf_crs.to_pickle(path)
+    result = pd.read_pickle(path)
+    assert_geodataframe_equal(result, gdf_crs)
+    geopandas.options.use_pygeos = True
```

Comment:
For some reason, this switch back to True is not picked on windows (if you remove it the same errors are in all envs). I have no clue why is that and no win machine to debug this. Since this is a super rare combination of factors (and temporary until shapely 2.0), would it be okay to skip this test under windows?

Reply:
Maybe you can try if using a setup/teardown with a fixture helps?

Something like
```
@pytest.fixture
def with_use_pygeos_false():
    orig = geopandas.options.use_pygeos
    geopandas.options.use_pygeos = no orig
    yield
    geopandas.options.use_pygeos = orig
```        

In theory it should do the same, but who knows